### ISSUE TITLE
Support for Optional when returning nullable types

### ISSUE THREAD

**Issue by paul-pop** [kobylynskyi/graphql-java-codegen#269]
Thank you for the library, it's great! 👏 

**Is your feature request related to a problem? Please describe.**
Because the query resolvers return the nullable object (without wrapping it in Optional) it can feel outdated for people heavily using Java8+ features.

**Describe the solution you'd like**
Currently when you have a query that has a nullable return type like this:

```graphql
type Query {
    user(id: ID!): User
}
```

the generated code is:

```java
@Override
public User user(final String id) {
    return null;
}
```

Would be nice to have a flag to enable/disable behaviour that allows the generation for nullable return types to use Optional instead which can make it easier to write neater code in the resolver.

```java
@Override
public User user(final String id) {
    return Optional.empty();
}
```

**Describe alternatives you've considered**
None

**Additional context**
None

**Comment by kobylynskyi** [comment]
@paul-pop thanks for reporting this feature request.
I've added a new config `useOptionalForNullableReturnTypes`: #274 
It will be released this week as part of version 3.0.0.

**Comment by kobylynskyi** [comment]
@paul-pop Version 3.0.0 was released today.
